### PR TITLE
Feate/create company fe

### PR DESCRIPTION
**Pull request by jimmacur**
## Type of Change
- [x] feature ⛲
- [x] styling 🎨
- [x] refactor 🧑‍💻
- [x] testing 🧪

## Description

✅ Summary of Added Functionality
	•	View Companies:
Ensures the Companies page displays a list of companies with correct details and handles cases where no companies are found.
	•	Search Companies:
Confirms the presence and functionality of a search bar for filtering companies.
	•	Add New Company:
Verifies that users can fill out a form to add new companies and that the submission works as expected.
	•	Adds comprehensive testing for viewing all companies, when no companies are found, adding new companies, and error handling for adding companies gone wrong.
	
	
In order to make this work you need to follow the instructions in the BE repo to get the token for a user. Then put that token between the quotes of line #29 of NewCompanies.tsx and #32 of Companies.tsx

## Motivation and Context

[#29](https://github.com/orgs/turingschool/projects/15/views/1?pane=issue&itemId=90046727&issue=turingschool%7Ctracker-crm%7C29)
[#24](https://github.com/orgs/turingschool/projects/15/views/1?pane=issue&itemId=89914372&issue=turingschool%7Ctracker-crm%7C24)

## Added Test?
- [x] Yes 🫡
  - 

## Checklist:

- [x] My code follows the code style of this project.

### CODE REVIEW

File: src/components/companies/Companies.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  const [companies, setCompanies] = useState<Company[] | null>([]); 
+  const [searchTerm, setSearchTerm] = useState("");
+  const [filteredCompanies, setFilteredCompanies] = useState<Company[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchCompanies = async () => {
+      try {
+        const token =
+          "eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoyLCJyb2xlcyI6WyJ1c2VyIl0sImV4cCI6MTczNDE4ODQ3Nn0.h2YmilmQWZ8CmopONeNZNZpiHIqtdR7M2Siw7jDISaU";
```
Comment (Sgalvin36) [inline src/components/companies/Companies.tsx:32]: Do we want to have a token in the code? Even if it isn't a valid one, I think its a potential security risk

File: src/components/companies/NewCompany.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      street_address: streetAddress,
+      city,
+      state,
+      zip_code: zipCode,
+      notes
+    };
+
+    try {
+      const token =
+        "eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoyLCJyb2xlcyI6WyJ1c2VyIl0sImV4cCI6MTczNDE4ODQ3Nn0.h2YmilmQWZ8CmopONeNZNZpiHIqtdR7M2Siw7jDISaU";
```
Comment (Sgalvin36) [inline src/components/companies/NewCompany.tsx:29]: Same as previous, I'm not sure we want to have an example of what the token is in our code.

File: src/components/companies/NewCompany.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            <option value="SD">South Dakota</option>
+            <option value="TN">Tennessee</option>
+            <option value="TX">Texas</option>
+            <option value="UT">Utah</option>
+            <option value="VT">Vermont</option>
+            <option value="VA">Virginia</option>
+            <option value="WA">Washington</option>
+            <option value="WV">West Virginia</option>
+            <option value="WI">Wisconsin</option>
+            <option value="WY">Wyoming</option>
```
Comment (Sgalvin36) [inline src/components/companies/NewCompany.tsx:153]: This has potential for refactor. It gets the job done for now. Just something to take note of that we can dynamically generate html, then this could be pulled out into an object of "state: abbreviation" to generate all the options. Not necessary at this time since this already works, but in the event of a refactor, this can be one of the areas

File: src/components/layout/MenuBar.tsx
Comment (Sgalvin36) [inline src/components/layout/MenuBar.tsx:1]: Thank you for changing these all to links!

File: cypress/e2e/newCompanySpec.cy.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    cy.get('#website').type('www.testcompany.com').should('have.value', 'www.testcompany.com');
+    cy.get('#streetAddress').type('123 Test St').should('have.value', '123 Test St');
+    cy.get('#city').type('Test City').should('have.value', 'Test City');
+    cy.get('select').select('CO').should('have.value', 'CO');
+    cy.get('#zipCode').type('12345').should('have.value', '12345');
+    cy.get('textarea').type('Test notes').should('have.value', 'Test notes');
+  });
+
+  it('should require Company Name, Street Address, City, State, and Zip Code', () => {
+    cy.get('button[type="submit"]').click();
```
Comment (CCirbo) [inline cypress/e2e/newCompanySpec.cy.js:87]: Perhaps, and it doesn’t have to be on this PR, you could add some sad path tests like submitting an incomplete form to ensure validation works as expected, or simulating API failures to see how the app handles error scenarios gracefully.

**Comment by CCirbo** [comment]
I agree with Shane on the tokens being in our code; while I understand why it’s necessary right now due to login issues, I agree we should prioritize removing them in the future for security. I also commented earlier that adding sad path or edge case testing (e.g., submitting an incomplete form or handling API failures) would enhance the reliability of the tests, however, I think this could be addressed in a separate ticket to keep the scope of this PR focused.

This looks really solid, and merging it would be beneficial to start assembling the app flow while tracking the additional tasks for future improvements.

File: _site/tsconfig.json
Comment (jchirch) [inline _site/tsconfig.json:1]: I will second that praise for dealing with the `_site` clone files. Unsure of their origin, but that you for cleaning that up and ensuring the app still functions as expected.

File: cypress/e2e/companiesSpec.cy.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      body: mockCompanies,
+      headers: {
+        "Content-Type": "application/json"
+      }
+    }).as("getCompanies");
+    
+    cy.visit("http://localhost:3000/");
+    cy.get("img[alt='Companies']").click();
+    /*
+    cy.get("#email").type("danny_de@email.com")
```
Comment (jchirch) [inline cypress/e2e/companiesSpec.cy.js:16]: Are we keeping this commented out code as a future template or did it slip through the cracks? Not a huge deal either way, just putting it on everyone's radar that it needs to be removed prior to production. This filtering may warrant creating a new issue on the project board.

File: cypress/e2e/newCompanySpec.cy.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        zip_code: '12345',
+        notes: 'Test notes'
+      }
+    }).as('addCompany');
+    
+    cy.intercept("GET", "http://localhost:3001/api/v1/users/2/companies", {
+      statusCode: 200,
+      body: {
+        data: [
+          ...mockCompanies.data,
```
Comment (jchirch) [inline cypress/e2e/newCompanySpec.cy.js:27]: Good use of mocking (and a spread operator!), nice to get into the practice to save money and prevent API rate limits in future projects.

File: cypress/support/mockCompanies.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+const mockCompanies = {
```
Comment (jchirch) [inline cypress/support/mockCompanies.js:1]: Would mock companies be better off living in a fixture file? This works fine and seems to serve the same purpose, just curious on your reasoning on not making a new directory for all fixture files.

**Review by Sgalvin36 — APPROVED** [review]
Thanks for the discussion on this!

**Comment by jimmacur** [comment]
Great idea @CCirbo. I will take the codes out and leave the instructions to get the token and where to put it in the PR description.
